rfc/website-revamp.md

```diff
@@ -0,0 +1,143 @@
+* Work completed by July 1.
+
+## Design
+
+### Spinnaker Personas
+* Individual Contributor: I am a developer who contributes code to the core Spinnaker project, or maintains a fork or custom extensions of Spinnaker for my company.
+* Operator: I manage a Spinnaker installation for my company.
+* App Developer: I am on an application team and use Spinnaker to manage my continuous delivery workflow as an end-user.
+* Non-technical stakeholder/Executive: I am evaluating Spinnaker as a means to achieve some business objective.
+
+### Top Nav Menu
+The current top nav will be condensed into 5 tabs:
+
+* Getting Started
+    * Basic conceptual info about Spinnaker for multiple personas.
+    * Direct people to the next logical step for their persona
+* Docs
+    * All the current Spinnaker product documentation
+* Community
+    * Contributor Experience Resource, such as code and documentation contribution guide and contribution best practices and tips
+    * Community Calendar
+    * Usage stats
+    * Governance & SIG info. This should be autogenerated from the Governance repo if possible. If not, we will continue to link to the Governance repo.
+* News
+    * Spinnaker newsletter
+    * Featured videos
+* Blog
+
+### Entrypoints & Tracks:
+Check out [the mock-up draft](https://docs.google.com/drawings/d/1FCzzvnK0a-_dQ5Nn1T4jz4DrXfr4tMzc9BrObMdEw3k/edit) to visualize the proposed content.
+
+*Note: tracks overlap just as personas overlap for real individuals, based on roles and experience*
+
+* Executive track: Should answer "What does my life look like if my team(s) use Spinnaker?" for Executives
+  * Card prompting clicks from those seeking to achieve business objectives, "Accelerate the velocity of your engineering teams"
+    * leads to a persona-tailored content link in Getting Started
+  * Case study content cards
+  * Who should use Spinnaker card
+
+* App Developer track: Should answer "What does my life look like if I use Spinnaker?" for AppDevs
+  * Card prompting appdevs to, "Take your code from commit to production in one pipeline"
+    * leads to a persona-tailored content link in Getting Started
+    * Video carousel
+    * Spinnaker Universe Calendar card
+    * Slack join card
+
+* Operator track: Should answer "What does my life look like if I use Spinnaker?" for Operators
+  * Card prompting operators to, "Create and operationalize a golden path to production for AppDevs"
+    * leads to a persona-tailored content link in Getting Started
+    * Video carousel
+    * Spinnaker Universe Calendar card
+    * Slack join card
+
+* Individual Contributor track: Should answer "What are my next steps?" for potential contributors
+  * Contributor card
+    * leads to Contributor section in the Community area
+  * Spinnaker Universe Calendar card
+
+### Documentation Features
+
+* Versioned documentation that corresponds to Spinnaker versions. This is a small widget in the top nav. See [K8s docs](kubernetes.io/docs/home) for an example.
```

Comment:
I think versioned documentation would be great, as we often just add inline to the documentation "you must have Spinnaker 1.x.y to use this feature", but I'd like to understand a bit better how this would be technically implemented.

I think versioned documentation is easiest in cases where the docs are auto-generated by the code, but most of our docs are not.  What would the experience look like for someone writing docs going forward?  Would there be multiple branches in the docs site and contributors would need to update the relevant branches depending on the docs they are writing?  (That is, if you're adding missing documentation for an old feature, you'd need to cherry pick that back, or something similar?)

Reply:
Proprietary documentation w/ versions typically maintain all versions that haven't been deprecated. K8s maintains only the most recent minor version, but the previous versions are still available for people to reference. That seems like a small but useful first step for us. 

I'll add the details on how we would implement this in the RFC.